Add removing of leading whitespace and all punctuation in title_sort

danamansana opened the pull request:
This PR addresses SCC-3111. We need to make sure that `"Five Itchy Aardvarks"` gets sorted with `F` instead of before `A`

This PR:

- Adds a utility function to remove leading spaces, as well as all punctuation, from a string
- Calls this function when generating `title_sort`
- Adds tests

nonword reviewed (commented):
There was additional requirements gathering happening as you created this PR. Can you check [the ticket](https://jira.nypl.org/browse/SCC-3111) and consider implementing the rest of the transformations? Clearly the timing was unfair, so if we need to generate a different ticket, that's understandable.

danamansana commented:
@nonword  I've added some more processing which I think should take care of all the additional requirements. I've renamed the method to something a little more generic since now it is doing more than just removing punctation.

nonword reviewed (approved):
Lovely
